src/parser/Parser.cpp

```diff
@@ -3,98 +3,105 @@
 Parser::Parser(const Lexer lexer)
     : m_lexer(lexer)
 {
+    m_currenttoken = m_lexer.getToken();
```

Comment:
Can we do camel case? I don't like camel case that much either but everything else in the frontend is camel case already.

Also personally I would prefer a shorter name like `m_currTok` but I spend 8 hours a day reading Java boilerplate so that might just be me 😛